### PR TITLE
Complete overhall and simplification

### PR DESCRIPTION
**Pull request by nyukuru**
quick edit to be mod friendly because it was brought up today lol

### CODE REVIEW

**Review by crashb — APPROVED** [review]
Awesome, thanks! Before you merge this, I have 2 items I'd like to double-check:
1. Does the newly-introduced dependency on the "Components/asl-help" file mean that the user needs to manually download this file when installing the autosplitter?
2. Have you tested that this is working on both vanilla and modded copies of the game?

**Comment by nyukuru** [comment]
> Awesome, thanks! Before you merge this, I have 2 items I'd like to double-check:
> 
>     1. Does the newly-introduced dependency on the "Components/asl-help" file mean that the user needs to manually download this file when installing the autosplitter?
> 
>     2. Have you tested that this is working on both vanilla and modded copies of the game?

You will need to add a line in LiveSplit.AutoSplitters.xml such that it will download the binary (https://github.com/just-ero/asl-help/raw/main/lib/asl-help) along with the asl. There are plenty examples of this already on the xml if needed. 

It's tested and works fine both modded and vanilla due to the nature of how the fields are fetched, I am weary however that the fields I have found aren't the same as the ones you used (given there's multiple that do similar things) in the previous splitter so if you do notice that please do let me know.

File: neonwhite.asl [hunk 1]
```diff
@@ -1,99 +1,144 @@
...
+gameTime 
+{
+    if (settings["levelRush"])
+    {
+        long totalMilliseconds = (current.levelRushMicroseconds + current.levelPlaythroughMicroseconds) / 1000;
+        return TimeSpan.FromMilliseconds(totalMilliseconds);
+    }
+    else if (settings["chapterRush"])
+    {
+        return 0;
```
Comment (ero-qt) [inline neonwhite.asl]: This should error. Consider using `TimeSpan.Zero`.

**Comment by crashb** [comment]
Okay, I added the line in LiveSplit.AutoSplitters.xml (see [here](https://github.com/LiveSplit/LiveSplit.AutoSplitters/pull/1706)). Please confirm that this looks right.

Two more things:
1. I just tested the autosplitter and it added some numbers to the bottom of the layout. Do you know why this is happening? I don't see anything in the ASL file that would obviously cause it, so maybe it's because of asl-help?
image
2. The options you added that aren't currently used by the autosplitter should be clearly marked as "UNIMPLEMENTED," otherwise, we will get bug reports for options that haven't been implemented yet ;)

File: neonwhite.asl [hunk 2]
```diff
@@ -1,99 +1,144 @@
...
+            {"levelRush", "lr-sp-l", true, "Split timer when completing a level."},
+            {"levelRush", "lr-sp-c", true, "Split timer when level rush completed."},
+        { null, "chapterRush", false, "Chapter Rush"},
+            {"chapterRush", "cr-st", true, "Start timer when loading into a level from level select."},
+            {"chapterRush", "cr-re", true, "Reset timer when exiting back to level select."},
+            {"chapterRush", "cr-sp", true, "Split timer when completing a level."},
+        { null, "newGame", false, "New Game"},
+            {"newGame", "ng-st", true, "Start timer when pressing \"New Game\"."},
+            {"newGame", "ng-sp-l", true, "Split timer when completing a level."},
+            {"newGame", "ng-sp-g", true, "Split timer when collecting a present."}
```
Comment (ero-qt) [inline neonwhite.asl]: Are these used?

**Comment by ero-qt** [comment]
> 1. I just tested the autosplitter and it added some numbers to the bottom of the layout. Do you know why this is happening? I don't see anything in the ASL file that would obviously cause it, so maybe it's because of asl-help?

https://github.com/crashb/neon-white-autosplitter/blob/646406e1a2c03bc72d46078404dec239efbc826b/neonwhite.asl#L90-L91

**Comment by nyukuru** [comment]
Sorry, I meant to make that WIP commit in a separate branch but was in a rush. I decided to just go ahead and implement everything and did some quick tests to make sure it all works.

**Pull request merged**